bin/plugin/commands/performance.js

```diff
@@ -393,32 +403,39 @@ async function runPerformanceTests( branches, options ) {
 
 	/** @type {Record<string,Record<string, WPPerformanceResults>>} */
 	const results = {};
+	const wpEnvPath = path.join(
+		performanceTestDirectory,
+		'node_modules/.bin/wp-env'
+	);
+
 	for ( const testSuite of testSuites ) {
 		results[ testSuite ] = {};
 		/** @type {Array<Record<string, WPPerformanceResults>>} */
 		const rawResults = [];
-		// Alternate three times between branches.
 		for ( let i = 0; i < TEST_ROUNDS; i++ ) {
+			const roundInfo = `round ${ i + 1 } of ${ TEST_ROUNDS }`;
+			log( `    >> Suite: ${ testSuite } (${ roundInfo })` );
 			rawResults[ i ] = {};
 			for ( const branch of branches ) {
-				const runKey = `${ branch }_${ testSuite }_run-${ i }`;
+				const runKey = `${ testSuite }_${ branch }_run-${ i }`;
```

Comment:
is there context behind this swap? not the most important thing, but it stands out in this PR where we're working on the path changes.

I guess there's a claim that grouping by test suite is more favorable than grouping by branch?

Reply:
I've swapped it because [it's how we actually iterate through those tests](https://github.com/WordPress/gutenberg/blob/4538099f41f81783720bc9c6d619bf06002ce79e/bin/plugin/commands/performance.js#L411-L442) and how we compare the results (`suite A: branch X vs. branch Y`, `suite B: branch X vs. branch Y`, etc.). So both the log info and file structure will now be compiled in this fashion:

| runner info (current) | runner info (new) |
| ------------- | ------------- |
| <img width="592" alt="Screenshot 2023-03-10 at 11 47 34" src="https://user-images.githubusercontent.com/1451471/224313846-8d2f84c0-16f8-4616-99ec-29bed5478ac1.png"> | <img width="418" alt="Screenshot 2023-03-10 at 11 47 55" src="https://user-images.githubusercontent.com/1451471/224313849-368fcedb-6f4c-4937-b977-7f1e3f1ea435.png"> |

| file structure (current) | file structure (new) |
| ------------- | ------------- |
| <img width="674" alt="Screenshot 2023-03-10 at 11 44 42" src="https://user-images.githubusercontent.com/1451471/224310700-85386795-60d1-4ef7-8a43-93827dc60f8f.png"> | <img width="652" alt="Screenshot 2023-03-10 at 11 45 03" src="https://user-images.githubusercontent.com/1451471/224310707-c55f586e-6883-4637-a35c-e3050393e9a9.png"> |

Reply:
As a side note, I think we should also replace `trunk` in the reference measurements with commit SHA, because `trunk` alone doesn't say much, IMO. Also, the summary should point to what is the reference and what is subject of the comparison, and display a 3rd delta column to show whether something got slower or faster, for example:

```
┌──────────────────────┬─────────────────────────┬────────────────────────┬─────────────────┐
│       (index)        │    trunk (reference)    │    32b4c12 (subject)   │        Δ        │
├──────────────────────┼─────────────────────────┼────────────────────────┼─────────────────┤
│    serverResponse    │       '235.12 ms'       │      '233.46 ms'       │      -0.5%      │
│      firstPaint      │       '251.88 ms'       │      '186.88 ms'       │     +18.2%      │
│   domContentLoaded   │       '685.96 ms'       │      '651.18 ms'       │      +0.5%      │
│        loaded        │       '686.72 ms'       │      '652.04 ms'       │      x.xx%      │
│ firstContentfulPaint │       '8436.28 ms'      │       '8057 ms'        │      x.xx%      │
│      firstBlock      │       '9095.38 ms'      │      '8713.32 ms'      │      x.xx%      │
│         type         │        '42.98 ms'       │       '43.03 ms'       │      x.xx%      │
│       minType        │        '41.9 ms'        │       '41.92 ms'       │      x.xx%      │
│       maxType        │        '45.71 ms'       │       '46.52 ms'       │      x.xx%      │
│    typeContainer     │        '14.81 ms'       │       '14.99 ms'       │      x.xx%      │
│   minTypeContainer   │        '13.35 ms'       │       '13.69 ms'       │      x.xx%      │
│   maxTypeContainer   │        '18.87 ms'       │       '17.7 ms'        │      x.xx%      │
│        focus         │        '40.99 ms'       │       '44.78 ms'       │      x.xx%      │
│       minFocus       │        '35.64 ms'       │       '35.16 ms'       │      x.xx%      │
│       maxFocus       │        '64.35 ms'       │       '71.31 ms'       │      x.xx%      │
│     inserterOpen     │        '33.11 ms'       │       '32.39 ms'       │      x.xx%      │
│   minInserterOpen    │        '27.96 ms'       │       '27.73 ms'       │      x.xx%      │
│   maxInserterOpen    │        '52.56 ms'       │       '52.9 ms'        │      x.xx%      │
│    inserterSearch    │        '13.53 ms'       │       '12.29 ms'       │      x.xx%      │
│  minInserterSearch   │        '5.76 ms'        │       '6.11 ms'        │      x.xx%      │
│  maxInserterSearch   │        '19.31 ms'       │       '18.43 ms'       │      x.xx%      │
│    inserterHover     │        '26.31 ms'       │       '26.4 ms'        │      x.xx%      │
│   minInserterHover   │        '22.51 ms'       │       '22.38 ms'       │      x.xx%      │
│   maxInserterHover   │        '47.75 ms'       │       '46.44 ms'       │      x.xx%      │
│     listViewOpen     │       '157.32 ms'       │      '159.09 ms'       │      x.xx%      │
│   minListViewOpen    │       '142.89 ms'       │      '142.65 ms'       │      x.xx%      │
│   maxListViewOpen    │       '235.52 ms'       │      '253.54 ms'       │      x.xx%      │
└──────────────────────┴─────────────────────────┴────────────────────────┴─────────────────┘
```

What do you think?

Reply:
> I think we should also replace trunk in the reference measurements with commit SHA

if we can display both that also has value. I think different needs use them differently. for most purposes `trunk` does a better job communicating than `13a14ca` because we recognize that when a CI job runs it's running against the latest `trunk` at the time the CI job ran. we can always lookup what `13a14ca` is, but that involves additional steps when `trunk` is pretty bare and simple.

> what is the reference and what is subject of the comparison

for most runs this matters, but the script supports multiple "branches" beyond two, and in those cases I'm not sure one is the reference. if we include this we will be making a decision that _one_ of the "branches" is the reference, likely the first one. (branches is in quotes because it's really just a `git` ref).

> and display a 3rd column (delta) to show whether we're slowing down or speeding up, for example:

no strong opinion here on what we _should_ do, but I can share my personal hesitancy with this due to the fact that we're still hitting some wild variation across tests (even with the recent fixes), and percentages always make changes look simpler than they are. I like making the easy-to-misuse things harder to use, whereas simply reporting `ms` is just that, a fact reported without any interpretation, because the interpretation is itself extremely difficult to automate.